### PR TITLE
[IMP] mrp: mrp_document

### PR DESCRIPTION
**Pull request by anr-odoo**
Task : https://www.odoo.com/web#id=1960733&action=327&model=project.task&view_type=form&menu_id=4720

Pad :
https://www.odoo.com/web#id=1960733&action=327&model=project.task&view_type=form&menu_id=4720

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp/static/src/js/mrp_documents_controller_mixin.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+     */
+    _onKanbanPreview(ev) {
+        ev.stopPropagation();
+        const documents = ev.data.recordList;
+        const documentID = ev.data.recordID;
+        const documentViewer = new DocumentViewer(this, documents, documentID);
+        documentViewer.appendTo(this.$('.o_mrp_documents_kanban_view'));
+    },
+    /**
+     * specially created to call _processFiles method from tests
```
[the remark this answers is not in the document]
Comment (msh-odoo) [reply addons/mrp/static/src/js/mrp_documents_controller_mixin.js]: @svs-odoo This comment seems before fileUploadMixin introduced, we now use fileUplodMixin so yes this comment should be changed, instead of ```_processFiles``` it should ```_uploadFiles```

[the remark this answers is not in the document]
Comment (anr-odoo) [reply addons/mrp/static/src/js/mrp_documents_controller_mixin.js]: @svs-odoo @msh-odoo comment has been changed.

File: addons/mrp/views/ir_attachment_view.xml [hunk 2]
```diff
@@ -14,9 +14,12 @@
                 <field name="create_uid"/>
                 <templates>
                     <t t-name="kanban-box">
+                        <t t-set="fileRequest" t-value="record.type.raw_value === 'empty'"/>
                         <div class="oe_kanban_global_area o_kanban_attachment oe_kanban_global_click">
                             <div class="o_kanban_image">
-                                <div class="o_kanban_image_wrapper">
+                                <t t-set="binaryPreviewable"
+                                   t-value="new RegExp('(image|video|application/pdf|text)').test(record.mimetype.value) &amp;&amp; record.type.raw_value === 'binary'"/>
+                                <div t-attf-class="o_kanban_image_wrapper #{(webimage or binaryPreviewable) ? 'o_kanban_previewer' : ''}" t-att-title="fileRequest ? 'Upload file' : ''">
```
Comment (svs-odoo) [inline addons/mrp/views/ir_attachment_view.xml]: @anr-odoo Why this change ? Can we have a file request with mrp documents ? 
Also, please don't push anything from now, I rework your branch to merge it (docstring, guidelines, ...).

Comment (anr-odoo) [reply addons/mrp/views/ir_attachment_view.xml]: @svs-odoo  By mistake change was there something gets messed in rebasing.

**Comment by sle-odoo** [comment]
robodoo r+ rebase-ff

**Comment by sle-odoo** [comment]
robodoo r-

**Comment by sle-odoo** [comment]
robodoo r+ rebase-ff
